### PR TITLE
Modify rule S6234 Improve rule specification (CPP-3713)

### PR DESCRIPTION
**Pull request by loic-joly-sonarsource**

### CODE REVIEW

File: rules/S6234/cfamily/rule.adoc [hunk 1]
```diff
@@ -1,40 +1,37 @@
 ``++auto++`` is a type placeholder that may be used in variable declarations to instruct the compiler to infer the type from the initializer.
 
 
-The use of ``++auto++`` reduces unnecessary boilerplate in the situation when the type of the variable is apparent from the context. However, outside of some obvious situations (e.g. when type is spelled in the initializer), the judgment is hard to make. Usually, it is trying to balance between the verbosity of type and the readability from the contributor's perspective.
+The use of ``++auto++`` reduces unnecessary boilerplate in the situation when the type of the variable is apparent from the context. In other situations, whether `auto` increases or decreases readability is a matter of personal taste.
```
Comment (amelie-renard-sonarsource) [inline rules/S6234/cfamily/rule.adoc:4]:
```suggestion
The use of ``++auto++`` reduces unnecessary boilerplate in situations where the type of the variable is apparent from the context. In other situations though, whether `auto` increases or decreases readability is a matter of personal taste.
```

File: rules/S6234/cfamily/rule.adoc [hunk 2]
```diff
@@ -1,40 +1,37 @@
...
 
 
-Arguably, in the case of the variables initialized from a function that conventionally returns an iterator (e.g. ``++begin++``, ``++end++``, ``++std::find++``), it is clear that the type of the variable is some iterator. Spelling the exact type of the iterator in such a situation does not improve the clarity of the code, especially considering their usual verbosity.
+Arguably, in the case of the variables initialized from a function that conventionally returns an iterator (e.g. `begin`, `end`, `std::find`), it is clear that the type of the variable is some iterator. Spelling the exact type of the iterator in such a situation does not improve the clarity of the code, especially considering the usual verbosity of such types.
```
Comment (amelie-renard-sonarsource) [inline rules/S6234/cfamily/rule.adoc:7]: "Arguably" seems strange to me, specially when it is written "it is clear" just after.
"Unarguably"? Or more simply "However"?

**Comment by loic-joly-sonarsource** [comment]
@amelie-renard-sonarsource  This PR was erroneously left open for quite some time... I made it up to date, and think it can be merged (I changed the Jira ticket number because the initial ticket was closed)

File: rules/S6234/cfamily/rule.adoc [hunk 3]
```diff
@@ -1,41 +1,39 @@
 ``++auto++`` is a type placeholder that may be used in variable declarations to instruct the compiler to infer the type from the initializer.
 
 
-The use of ``++auto++`` reduces unnecessary boilerplate in the situation when the type of the variable is apparent from the context. However, outside of some obvious situations (e.g. when type is spelled in the initializer), the judgment is hard to make. Usually, it is trying to balance between the verbosity of type and the readability from the contributor's perspective.
+The use of ``++auto++`` reduces unnecessary boilerplate in situations where the type of the variable is apparent from the context (see rule S5827). In other situations, though, whether `auto` increases or decreases readability is a matter of personal taste.
```
Comment (amelie-renard-sonarsource) [inline rules/S6234/cfamily/rule.adoc:4]:
```suggestion
The use of `auto` reduces unnecessary boilerplate in situations where the type of the variable is apparent from the context (see rule S5827). In other situations, though, whether `auto` increases or decreases readability is a matter of personal taste.
```

**Review by amelie-renard-sonarsource — APPROVED** [review]

**Pull request merged**